### PR TITLE
SE-1186 Enhances the analytics_single.yml playbook

### PR DESCRIPTION
**Pull request by pomegranited**
Modifies the `analytics_single.yml` playbook to be able to use it to set up single-VM production analytics servers.

**JIRA tickets**: [OSPR-3695](https://openedx.atlassian.net/browse/OSPR-3695)

**Sandbox URL**:

* Insights: https://se-1186-analytics-test.sandbox.opencraft.hosting
* Analytics API: https://se-1186-analytics-test.sandbox.opencraft.hosting:4330

**Merge deadline**: None

**Testing instructions**:

We haven't set up DNS for this server, so you'll need an `/etc/hosts` entry to map the IP address to the hostname:
```
54.38.245.235 se-1186-analytics-test.sandbox.opencraft.hosting
```

1. Ensure that Insights is running and accessible over SSL.
   (There's only a self-signed certificate, and authentication won't work, since OAuth has not been set up to a running edxapp instance.)
1. Ensure that the analytics API is running and accessible over SSL.
   (Also using a self-signed certificate)
   
**Author Notes & Concerns**

1. We had issues getting the `analytics_pipeline` role to run in our sandbox environment because it's not running a local mysql instance.  The role tries to [create mysql users](https://github.com/open-craft/configuration/blob/jill/analytics_server/playbooks/roles/analytics_pipeline/tasks/main.yml#L223-L231), but isn't configured to connect to mysql on a different host, so throws: 
   ```
   FAILED! => {"changed": false, "msg": "unable to connect to database, check login_user and login_******** are correct or /root/.my.cnf has the credentials. Exception message: (2002, \"Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)\")"}
   ```
   Not sure what uses this role?  Our use case only requires `insights` and the `analytics_api` to be running, so we disabled the pipeline with: `SANDBOX_ENABLE_ANALYTICS_PIPELINE: false`

**Reviewers**
- [ ] @gr4scale
- [ ] edX reviewer[s] TBD

Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR if it is code shared across multiple services and you don't own all of the services.
  - [x] Are you adding any new default values that need to be overridden when this change goes live? No
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).
  - [ ] Think about how this change will affect Open edX operators.  Have you updated the wiki page for the next Open edX release?

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @pomegranited! I've created [OSPR-3695](https://openedx.atlassian.net/browse/OSPR-3695) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by feanil** [comment]
Thanks for the PR @pomegranited I'll queue this up for review.  FYI @macdiesel

**Comment by pomegranited** [comment]
Thanks @feanil @macdiesel  -- will ping you when we've got this tested and approved internally.

**Comment by gr4yscale** [comment]
@pomegranited 
:+1:

Nice work :) And it's good that you discovered the mysql user creation issue, so at least it's documented.

- [x] I tested this: I confirmed that both Insights and Analytics API services are accessible on your sandbox, with SSL
- [x] I read through the code
- [ ] I checked for accessibility issues **N/A**
- [x] Includes documentation
- [ ] I made sure any change in configuration variables is reflected in the corresponding
    client's `configuration-secure` repository. **N/A**

**Comment by pomegranited** [comment]
@feanil @macdiesel This is ready for your review.  Thank you!

File: playbooks/analytics_single.yml [hunk 1]
```diff
@@ -7,14 +18,31 @@
     disable_edx_services: false
     ENABLE_DATADOG: False
     ENABLE_NEWRELIC: False
+    SANDBOX_ENABLE_ANALYTICS_API: True
+    SANDBOX_ENABLE_ANALYTICS_PIPELINE: True
```
Comment (pomegranited) [inline playbooks/analytics_single.yml:22]: The `analytics_pipeline` role [creates users on the mysql database](https://github.com/open-craft/configuration/blob/jill/analytics_server/playbooks/roles/analytics_pipeline/tasks/main.yml#L223-L231), which only works if mysql is run locally and has the admin credentials available on the server.

Would it be ok to disable this variable and role by default here?

**Comment by pomegranited** [comment]
@feanil @macdiesel  CC @natabene  Do you know when you might have time to review this change?

**Review by macdiesel — APPROVED** [review]
@pomegranited I've been on vacation. Looks good. Are you ready to merge?

**Comment by pomegranited** [comment]
Thanks @macdiesel , yep!  We've redeployed analytics for a client with this updated playbook, so it's good to squash and merge.

**Comment by openedx-webhooks** [comment]
@pomegranited 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/configuration&entry.867055334=https://github.com/edx/configuration/pull/5223&entry.1484655318=https://github.com/pomegranited&entry.752974735=2019-06-26+06:24&entry.1917517419=2019-07-19+00:51&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**